client/my-sites/comments/comment/comment-author-more-info.jsx

```diff
@@ -3,33 +3,253 @@
+
+		updateBlacklist( siteId, newBlacklist, analytics );
+
+		showNotice(
+			translate( 'User %(email)s is blocked and can no longer comment on your site.', {
+				args: { email: authorEmail },
+			} ),
+			noticeOptions
+		);
+	};
 
 	render() {
+		const {
+			authorDisplayName,
+			authorEmail,
+			authorIp,
+			authorUrl,
+			authorUsername,
+			isAuthorBlacklisted,
+			showBlockUser,
+			siteSlug,
+			trackAnonymousModeration,
+			translate,
+		} = this.props;
+
+		const { showPopover } = this.state;
+
 		return (
-			<div
-				className="comment__author-more-info"
-				onClick={ this.openAuthorMoreInfoPopover }
-				ref={ this.storeLabelRef }
-			>
-				<InfoPopover ignoreContext={ this.authorMoreInfoLabel } ref={ this.storePopoverRef }>
-					Author More Info
-				</InfoPopover>
-				<label>User Info</label>
+			<div className="comment__author-more-info">
+				<Button borderless onClick={ this.togglePopover } ref={ this.storePopoverButtonRef }>
+					<Gridicon icon="info-outline" size={ 18 } />
+					{ translate( 'User Info' ) }
+				</Button>
+
+				<Popover
+					className="comment__author-more-info-popover"
+					context={ this.popoverButton }
+					isVisible={ showPopover }
+					onClose={ this.closePopover }
+					position="bottom"
+				>
+					<div className="comment__author-more-info-element">
+						<Gridicon icon="user-circle" />
+						<div>
+							<div>
+								<strong>{ authorDisplayName || translate( 'Anonymous' ) }</strong>
+							</div>
+							<div>{ authorUsername }</div>
+						</div>
+					</div>
+
+					<div className="comment__author-more-info-element">
+						<Gridicon icon="mail" />
+						<div>{ authorEmail || <em>{ translate( 'No email address' ) }</em> }</div>
+					</div>
+
+					<div className="comment__author-more-info-element">
+						<Gridicon icon="link" />
+						<div>
+							{ !! authorUrl && (
+								<ExternalLink href={ authorUrl }>
+									<Emojify>{ urlToDomainAndPath( authorUrl ) }</Emojify>
+								</ExternalLink>
+							) }
+							{ ! authorUrl && <em>{ translate( 'No website' ) }</em> }
+						</div>
+					</div>
+
+					<div className="comment__author-more-info-element">
+						<Gridicon icon="globe" />
+						<div>{ authorIp || <em>{ translate( 'No IP address' ) }</em> }</div>
+					</div>
+
+					{ showBlockUser && (
+						<div className="comment__author-more-info-element">
+							<Button onClick={ this.toggleBlockUser } scary={ ! isAuthorBlacklisted }>
+								{ isAuthorBlacklisted ? translate( 'Unblock user' ) : translate( 'Block user' ) }
+							</Button>
+						</div>
+					) }
+
+					{ ! authorEmail && (
+						<div className="comment__author-more-info-element">
+							<div>
+								{ translate(
+									// eslint-disable-next-line max-len
+									"Anonymous messages can't be blocked individually, but you can update your {{a}}settings{{/a}} to only allow comments from registered users.",
```

Comment:
Did we run into trouble with translate doing a string concat here?

Reply:
It never occurred to me that `translate` accepted concatenated strings!